### DISCUSSION TITLE
args

### DISCUSSION THREAD

**Discussion by LiuPengPeter** [idaholab/moose#16321] [Q&A Modules: General] [answered]
Hello guys,

The args is related with computeQpJacobian and  computeQpOffDiagJacobian. Sometimes I could make sure which variable is needed. Could I write all of variables except the "variable" ?

**Comment by lindsayad** [comment]
I don't understand your question. Can you add some more context/information please?

**Reply by LiuPengPeter** [reply]
For example, my kernel is F*c.
If F is the function of eta1, c1, eta2 and c2, then the args= 'eta1 c1 eta2 c2'.
If F is the function of c1 and c2, then the args= 'c1 c2'.
Sometimes I forget to change, so I wonder if I could write all of them at first even if some won't be used.

**Comment by dschwen** [comment, marked as answer]
That's not recommended as it will result in a combinatorial explosion of material property derivatives being requested. Properly written objects that require an `args` parameter should warn you if you are missing important variables in that list.

**Reply by LiuPengPeter** [reply]
Thank you.
I understand now. With all of variables added, it won't lead wrong result but the workload of simulation will be very large.
